staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go

```diff
@@ -40,15 +51,22 @@ func WithAuthorization(handler http.Handler, requestContextMapper request.Reques
 			responsewriters.InternalError(w, req, errors.New("no context found for request"))
 			return
 		}
+		ae := request.AuditEventFrom(ctx)
 
 		attributes, err := GetAuthorizerAttributes(ctx)
 		if err != nil {
 			responsewriters.InternalError(w, req, err)
 			return
 		}
 		authorized, reason, err := a.Authorize(attributes)
+		if len(reason) != 0 {
+			audit.LogAnnotation(ae, reasonAnnotationKey, reason)
+		} else if err != nil {
+			audit.LogAnnotation(ae, reasonAnnotationKey, "error")
```

Comment:
"internal error"?

Reply:
Make this a constant `reasonError = "internal error"`, and then you can use it in the test.

Reply:
Done.